centreon/www/class/centreon-clapi/centreonAPI.class.php

```diff
@@ -803,9 +803,13 @@ public function import($filename)
                 $i++;
                 $tab = preg_split('/;/', $string);
                 if (strlen(trim($string)) != 0 && !preg_match('/^\{OBJECT_TYPE\}/', $string)) {
-                    $this->object = trim($tab[0]);
-                    $this->action = trim($tab[1]);
-                    $this->variables = trim(substr($string, strlen($tab[0] . ";" . $tab[1] . ";")));
+                    $this->object = htmlspecialchars(trim($tab[0]), ENT_QUOTES, 'UTF-8');
+                    $this->action = htmlspecialchars(trim($tab[1]), ENT_QUOTES, 'UTF-8');
+                    $this->variables = htmlspecialchars(
+                        trim(substr($string, strlen($tab[0] . ";" . $tab[1] . ";"))),
+                        ENT_QUOTES,
+                        'UTF-8'
+                    );
```

Comment:
There are old 2015 code locations in this class which do exactly the same "protection".

Why not making this explicit through a dedicated method to avoid duplicate code everywhere and facilitate the tracking of this "protection" ?

Last remark : it will transform `"` into `'`. The double quote is a forbidden character inside a variable field ?

Reply:
Addendum, html entities will also introduce a semicolon `;` and change the "csv" format for the variable field.